pubsub/pulsar/pulsar.go

```diff
@@ -0,0 +1,132 @@
+package pulsar
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/apache/pulsar-client-go/pulsar"
+	"github.com/dapr/dapr/pkg/logger"
+
+	"github.com/dapr/components-contrib/pubsub"
+)
+
+const (
+	host             = "host"
+	subscriptionName = "subscriptionName"
+	enableTLS        = "enableTLS"
+)
+
+type Pulsar struct {
+	logger   logger.Logger
+	client   pulsar.Client
+	metadata pulsarMetadata
+}
+
+func NewPulsar(l logger.Logger) pubsub.PubSub {
+	return &Pulsar{logger: l}
+}
+
+func parsePulsarMetadata(meta pubsub.Metadata) (*pulsarMetadata, error) {
+	m := pulsarMetadata{}
+
+	if val, ok := meta.Properties[host]; ok && val != "" {
+		m.Host = val
+	} else {
+		return nil, errors.New("pulsar error: missing pulsar host")
+	}
+	if val, ok := meta.Properties[subscriptionName]; ok && val != "" {
```

Comment:
The Dapr runtime will automatically populate the metadata with a `consumerID` field. This should be used for the subscription name, so you can remove the subscription name from metadata.

See this example for the Kafka Pub/Sub:
https://github.com/dapr/components-contrib/blob/8eff31b19159e3438df98f5c60f6fe4c63c14527/pubsub/kafka/kafka.go#L229